pages/index.vue

```diff
@@ -1,20 +1,45 @@
 <template>
   <div class="p-4 w-full">
-    <repo-box v-for="repo in repos" :key="repo.id" :repo="repo"></repo-box>
+    <repo-box v-for="repo in filteredRepos" :key="repo.id" :repo="repo"></repo-box>
   </div>
 </template>
 
 <script>
 import RepoBox from '~/components/RepoBox.vue'
 import Repositories from '~/data/generated.json'
+import { mapState } from 'vuex'
+import { cloneDeep, get } from 'lodash'
 
 export default {
   components: {
     RepoBox
   },
-  data: function () {
-    return {
-      repos: Repositories
+  computed: {
+    ...mapState(['activeSortOption', 'isOrderByDesc']),
+    filteredRepos() {
+      const repositoriesData = cloneDeep(Repositories)
+      if (this.activeSortOption.id) {
+        repositoriesData.sort((a, b) => {
+          if (this.isOrderByDesc) {
+            return this.getRepoDataBySortOption(a) < this.getRepoDataBySortOption(b) ? 1 : -1
+          } else {
+            return this.getRepoDataBySortOption(a) > this.getRepoDataBySortOption(b) ? 1 : -1
+          }
+        })
+      }
+      return repositoriesData
+    }
```

Comment:
Since you're already storing the sorting order and function inside the VueX store, I'd suggest you to sort the list in the store itself with a getter. That way, you won't have to wastefully clone the repositories array every time a filter is applied.

Reply:
I have followed your suggestions and modified the code please review it again now